standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java

```diff
@@ -255,14 +260,22 @@ private void cleanUpZK() {
         @Cleanup
         CuratorFramework zclient = builder.build();
         zclient.start();
+
         for (String path : pathsTobeCleaned) {
+            CompletableFuture<String> future = new CompletableFuture<String>();
+            futures.add(future);
             try {
-                zclient.delete().guaranteed().deletingChildrenIfNeeded()
-                        .forPath(path);
+                zclient.delete().idempotent().deletingChildrenIfNeeded().inBackground(new BackgroundCallback() {
+                    @Override
+                    public void processResult(CuratorFramework client, CuratorEvent event) throws Exception {
+                        future.complete(path);
```

Comment:
Aren't we handling errors?

Is it possible that the CF is never completed neither successfully nor exceptionally?